app/controllers/assignments_controller.rb

```diff
@@ -36,28 +36,55 @@ def index
   end
 
   def new
-    @files = Assignment.new
-    @files.course_id = @course.id
-    @files.due_date = (Time.now + 1.week).end_of_day.strftime("%Y/%m/%d %H:%M")
-    @files.available = Time.now.strftime("%Y/%m/%d %H:%M")
-    @files.lateness_config_id = @course.lateness_config_id
-    @files.request_time_taken = true
+    last_assn = @course.assignments.order(created_at: :desc).first
+    if @assignment.is_a? Files
```

Comment:
If this fixes the loss of form data on error, then awesome.

If there's no way to put this in the model, then this is more evidence that there should be separate controllers when we next get to do a refactor pass.